### PR TITLE
Campaigns API: Return all rendered messages for a Campaign

### LINKED ISSUE #798: Support rendered param to campaigns API
Support passing a `rendered` query parameter to the GET `/v1/campaigns` endpoint to preview all possible messages for a Gambit Campaign, rendered with the corresponding Phoenix Campaign variables.

### PR DESCRIPTION
**Pull request by aaronschachter**
#### What's this PR do?
Adds a `messages` property to the GET `campaigns/:id` response, with keys corresponding to various Gambit message types, which we define as fields on our Contentful Campaign content type in Gambini. For each Contentful Campaign field name, we store an object with keys:
* `raw` -- the copy stored in Contentful to use
* `override` -- boolean value specifying whether the `raw` value is defined on the entry for the given Campaign ID, or the default Campaign
* `rendered` -- the rendered copy, replacing the Liquid style variable tags with values from the Phoenix Campaign.

Example response:
````
{
  "data": {
    "id": "7483",
    "title": "Rinse, Recycle, Repeat",
    "status": "active",
    "messages": {
      "gambitSignupMenuMessage": {
        "override": true,
        "raw": "Great - it's simple: Keep beauty and personal care products out of landfills by making fun and creative recycling bins for the bathroom! \n\nThis action should take between 10 - 20 mins. Make it colorful so friends and family won't forget to recycle their bathroom empties. \n\nWhen you're done, text START to share a photo of your bin and you'll be entered to win a $5000 scholarship!",
        "rendered": "Great - it's simple: Keep beauty and personal care products out of landfills by making fun and creative recycling bins for the bathroom! \n\nThis action should take between 10 - 20 mins. Make it colorful so friends and family won't forget to recycle their bathroom empties. \n\nWhen you're done, text START to share a photo of your bin and you'll be entered to win a $5000 scholarship!"
      },
      "externalSignupMenuMessage": {
        "override": true,
        "raw": "Thanks for joining Rinse Recycle Repeat!\n\nNearly half of Americans don’t regularly recycle their beauty and personal care products. That’s a major reason these items account for a significant amount of landfill waste.\n\nThe solution is simple: Make fun and creative bins for bathrooms.\n\nOnce you have created some bathroom recycling bins, take a pic to prove it! Then text {{cmd_reportback}} to share it with us!",
        "rendered": "Thanks for joining Rinse Recycle Repeat!\n\nNearly half of Americans don’t regularly recycle their beauty and personal care products. That’s a major reason these items account for a significant amount of landfill waste.\n\nThe solution is simple: Make fun and creative bins for bathrooms.\n\nOnce you have created some bathroom recycling bins, take a pic to prove it! Then text P to share it with us!"
      },
      "invalidSignupMenuCommandMessage": {
        "override": false,
        "raw": "Sorry, I didn't understand that.\n\nText {{cmd_reportback}} when you have {{rb_verb}} some {{rb_noun}}.\n\nIf you have a question, text {{cmd_member_support}}.",
        "rendered": "Sorry, I didn't understand that.\n\nText P when you have decorated some bins.\n\nIf you have a question, text Q."
      },
      "askQuantityMessage": {
        "override": false,
        "raw": "Sweet! First, what's the total number of {{rb_noun}} you {{rb_verb}}?\n\nSend the exact number back.",
        "rendered": "Sweet! First, what's the total number of bins you decorated?\n\nSend the exact number back."
      },
      "invalidQuantityMessage": {
        "override": false,
        "raw": "Sorry, that's not a valid number.\n\nWhat's the total number of {{rb_noun}} you have {{rb_verb}}?\n\nIf you have a question, text {{cmd_member_support}}.",
        "rendered": "Sorry, that's not a valid number.\n\nWhat's the total number of bins you have decorated?\n\nIf you have a question, text Q."
      },
      ....
````

#### How should this be reviewed?

**Local**
Pull this branch down and run Gambit locally. Verify that GET `/campaigns` doesn't return a `messages` object, but GET `/campaigns/:id` does.

**Staging**
Upon deploying this branch to staging, verify:
* http://ds-mdata-responder-staging.herokuapp.com/v1/campaigns
* http://ds-mdata-responder-staging.herokuapp.com/v1/campaigns/3590 -- no overrides
* http://ds-mdata-responder-staging.herokuapp.com/v1/campaigns/7483 -- contains overrides

#### Any background context you want to provide?
* This will be used by our [Slothbot](github.com/dosomething/slothbot) Slack app to allow staffers to preview all possible rendered messages Gambit will return for a Campaign.
* Avoiding rendering all messages on the GET campaigns index endpoint to help performance (and there aren't any apps that depend on the `messages` there)

#### Relevant tickets
Fixes #798

#### Checklist
- [x] Documentation added for new features/changed endpoints.
- [x] Tested on staging.

### CODE REVIEW

File: lib/contentful.js [hunk 1]
```diff
@@ -211,3 +214,60 @@ module.exports.renderMessageForPhoenixCampaign = function (phoenixCampaign, msgT
...
 };
+
+/**
+ * Returns all rendered messages for given phoenixCampaign, using defaults when no override defined.
+ */
+module.exports.renderAllMessagesForPhoenixCampaign = function (phoenixCampaign) {
+  logger.debug(`renderAllMessagesForPhoenixCampaign:${phoenixCampaign.id}`);
+
+  return new Promise((resolve, reject) => {
+    const campaignFieldNames = Object.keys(contentfulCampaignFields);
```
Comment (sergii-tkachenko) [inline lib/contentful.js]: contentfulCampaignFields?

Comment (sergii-tkachenko) [inline lib/contentful.js]: Ah, I see, it's file global. A little bit deceptive.

File: lib/contentful.js [hunk 2]
```diff
@@ -211,3 +214,60 @@ module.exports.renderMessageForPhoenixCampaign = function (phoenixCampaign, msgT
...
+    const campaignContent = {};
+    const result = {};
```
Comment (sergii-tkachenko) [inline lib/contentful.js]: Does it work with all asynchronous then calls?

Comment (aaronschachter) [inline lib/contentful.js]: @sergii-tkachenko Not sure what you're asking here -- this function's only called from within `fetchCampaign` in `routes/campaigns`...

**Review by sergiitk — APPROVED** [review]
👍

**Pull request merged**